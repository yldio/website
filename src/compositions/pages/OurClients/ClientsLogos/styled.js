import styled from 'styled-components';
import breakpoints from 'styles/breakpoints';
import { textBlue } from 'styles/colours';

export const Grid = styled.ul`
  display: grid;
  position: relative;
  grid-template-columns: 25% 25% 25% 25%;

  &last-child {
    width: 500px;
    background-color: #5e5cb2;
  }
  ${breakpoints.mobile`
  grid-template-columns: 33% 33% 33%;
  `};
  ${breakpoints.desktop`
  grid-template-columns: 25% 25% 25% 25%;
  `};
`;

export const Header = styled.div`
  color: ${textBlue};
  font-size: 2.25rem;
  text-align: center;
  margin-top: 85px;
  line-height: 1.44;
`;

export const Line = styled.p`
  width: 100px;
  height: 4px;
  background-color: #5e5cb2;
  text-align: center;
  margin: 11px auto 70px auto;
`;

export const LogoWrapper = styled.div`
  margin-left: auto;
  margin-right: auto;
  ${breakpoints.mobile`
    margin-bottom: 32px;
  `};
`;

export const Image = styled.img`
  max-width: 100%;
  top: 0;
  bottom: 0;
  margin: auto;
  width: 90%;
`;
