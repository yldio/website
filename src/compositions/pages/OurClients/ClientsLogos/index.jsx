import React from 'react';
import PageSection from 'components/PageSection';
import { Grid, Header, Line, LogoWrapper, Image } from './styled';
import Hive from '../../../../assets/hive.png';
import Economist from '../../../../assets/economist.png';
import Tate from '../../../../assets/tate.png';
import Discovery from '../../../../assets/discovery.png';
import Canon from '../../../../assets/canon.png';
import SamsungJoyen from '../../../../assets/samsung-joyent.png';
import Kingfisher from '../../../../assets/kingfisher.png';
import Trainline from '../../../../assets/trainline.png';
import Conde from '../../../../assets/conde-nas.png';
import Microsoft from '../../../../assets/microsoft.png';
import Newsuk from '../../../../assets/newsuk.png';
import Google from '../../../../assets/google.png';
import Thomascook from '../../../../assets/thomascookLogo.png';
import Compare from '../../../../assets/compare-market.png';
import Solera from '../../../../assets/solera.png';
import Skype from '../../../../assets/skype.png';
import Tes from '../../../../assets/tes.png';
import Perform from '../../../../assets/perform.png';
import Lamassu from '../../../../assets/lamassu.png';

const logos = [
  Hive,
  Economist,
  Tate,
  Discovery,
  Canon,
  SamsungJoyen,
  Kingfisher,
  Trainline,
  Conde,
  Microsoft,
  Newsuk,
  Google,
  Thomascook,
  Compare,
  Solera,
  Skype,
  Tes,
  Lamassu,
  Perform,
];

const ClientsLogos = index => (
  <PageSection>
    <Header>Who we work with</Header>
    <Line />
    <Grid>
      {logos.map(logo => (
        <LogoWrapper>
          <Image alt="" key={index} src={logo} />
        </LogoWrapper>
      ))}
    </Grid>
  </PageSection>
);

export default ClientsLogos;
