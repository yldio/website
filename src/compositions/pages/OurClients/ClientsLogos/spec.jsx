import React from 'react';
import { shallow } from 'enzyme';
import ClientsLogos from './index';

test('compositions/pages/OurClients/ClientsLogos snapshot', () => {
  const wrapper = shallow(<ClientsLogos />);

  expect(wrapper).toMatchSnapshot();
});
