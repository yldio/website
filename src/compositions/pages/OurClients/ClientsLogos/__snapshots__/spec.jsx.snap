// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compositions/pages/OurClients/ClientsLogos snapshot 1`] = `
<PageSection>
  <styled__Header>
    Who we work with
  </styled__Header>
  <styled__Line />
  <styled__Grid>
    <styled__LogoWrapper>
      <styled__Image
        alt=""
        key="[object Object]"
        src="/src/assets/hive.png"
      />
    </styled__LogoWrapper>
    <styled__LogoWrapper>
      <styled__Image
        alt=""
        key="[object Object]"
        src="/src/assets/economist.png"
      />
    </styled__LogoWrapper>
    <styled__LogoWrapper>
      <styled__Image
        alt=""
        key="[object Object]"
        src="/src/assets/tate.png"
      />
    </styled__LogoWrapper>
    <styled__LogoWrapper>
      <styled__Image
        alt=""
        key="[object Object]"
        src="/src/assets/discovery.png"
      />
    </styled__LogoWrapper>
    <styled__LogoWrapper>
      <styled__Image
        alt=""
        key="[object Object]"
        src="/src/assets/canon.png"
      />
    </styled__LogoWrapper>
    <styled__LogoWrapper>
      <styled__Image
        alt=""
        key="[object Object]"
        src="/src/assets/samsung-joyent.png"
      />
    </styled__LogoWrapper>
    <styled__LogoWrapper>
      <styled__Image
        alt=""
        key="[object Object]"
        src="/src/assets/kingfisher.png"
      />
    </styled__LogoWrapper>
    <styled__LogoWrapper>
      <styled__Image
        alt=""
        key="[object Object]"
        src="/src/assets/trainline.png"
      />
    </styled__LogoWrapper>
    <styled__LogoWrapper>
      <styled__Image
        alt=""
        key="[object Object]"
        src="/src/assets/conde-nas.png"
      />
    </styled__LogoWrapper>
    <styled__LogoWrapper>
      <styled__Image
        alt=""
        key="[object Object]"
        src="/src/assets/microsoft.png"
      />
    </styled__LogoWrapper>
    <styled__LogoWrapper>
      <styled__Image
        alt=""
        key="[object Object]"
        src="/src/assets/newsuk.png"
      />
    </styled__LogoWrapper>
    <styled__LogoWrapper>
      <styled__Image
        alt=""
        key="[object Object]"
        src="/src/assets/google.png"
      />
    </styled__LogoWrapper>
    <styled__LogoWrapper>
      <styled__Image
        alt=""
        key="[object Object]"
        src="/src/assets/thomascookLogo.png"
      />
    </styled__LogoWrapper>
    <styled__LogoWrapper>
      <styled__Image
        alt=""
        key="[object Object]"
        src="/src/assets/compare-market.png"
      />
    </styled__LogoWrapper>
    <styled__LogoWrapper>
      <styled__Image
        alt=""
        key="[object Object]"
        src="/src/assets/solera.png"
      />
    </styled__LogoWrapper>
    <styled__LogoWrapper>
      <styled__Image
        alt=""
        key="[object Object]"
        src="/src/assets/skype.png"
      />
    </styled__LogoWrapper>
    <styled__LogoWrapper>
      <styled__Image
        alt=""
        key="[object Object]"
        src="/src/assets/tes.png"
      />
    </styled__LogoWrapper>
    <styled__LogoWrapper>
      <styled__Image
        alt=""
        key="[object Object]"
        src="/src/assets/lamassu.png"
      />
    </styled__LogoWrapper>
    <styled__LogoWrapper>
      <styled__Image
        alt=""
        key="[object Object]"
        src="/src/assets/perform.png"
      />
    </styled__LogoWrapper>
  </styled__Grid>
</PageSection>
`;
